### DISCUSSION TITLE
How does the code calculate a 3D density for a 2D simulation?

### DISCUSSION THREAD

**Discussion by ghost** [epochpic/epoch#610] [Q&A] [answered]
Hello @Status-Mirror , 

In a 2D simulation we put the window dimensions, the number of points pe grid (nx, ny) and the number of particles per cell.
But we also put the density, which is in m^-3. If I understand correctly the density modifies the weight of the macropraticles so this is what matters in the end. But how does the code know to scale the third dimension for a 2D simulation? Is it n(2D)=n(3D)^2/3 ?
For example if you want to measure the charge of a surface S in 2D, would it be Q=e*S* n^2/3?

Thank you.

**Comment by DanRRRR** [comment, marked as answer]
The third, missing dimension in 2D vs 3D in the code assumed to be 1 m.

/* I also asked this question. 
Stuart, this question is candidate #1 for new paragraph in your "Getting Started"  called "Q & A" :)
